src/gravel/controllers/nodes/deployment.py

```diff
@@ -440,32 +477,29 @@ async def finish_bootstrap_cb(
                     msg=error
                 )
             await post_bootstrap_cb(success, error)
-            await _assimilate_devices()
 
-        async def _assimilate_devices() -> None:
-            logger.debug("deployment > assimilating devices")
-            self._progress = ProgressEnum.ASSIMILATING
             try:
-                orch = Orchestrator(self._gstate.ceph_mgr)
-                orch.assimilate_all_devices()
-
-                # wait a few seconds so the orchestrator settles down
-                await asyncio.sleep(5.0)
-                while not orch.all_devices_assimilated():
-                    await asyncio.sleep(1.0)
-
-                logger.debug("deployment > devices assimilated")
-                self._progress = ProgressEnum.DONE
-                await finisher(True, None)
-
-            except Exception as e:
+                await _assimilate_devices()
+            except DeploymentError as e:
+                logger.error("unable to assimilate devices")
+                logger.exception(e)
                 self._state.mark_error(
                     code=DeploymentErrorEnum.CANT_ASSIMILATE,
-                    msg=str(e)
+                    msg=e.message
                 )
-                logger.error("unable to assimilate devices")
-                logger.exception(e)
-                await finisher(False, str(e))
+                await finisher(False, e.message)
+            else:
+                self._progress = ProgressEnum.DONE
+                await finisher(True, None)
+
+        async def _assimilate_devices() -> None:
```

Comment:
(nit) Might want to move this to before `finish_bootstrap_cb` since it is called from within there and scoping gets confusing with the identically named methods on the class.

Alternatively (and personal preference) this doesn't need to be a method and could just be inside of the above `try` statement for clarity.